### PR TITLE
ipn/{ipnserver,localapi}: fix InUseOtherUser handling with WatchIPNBus

### PR DESCRIPTION
**Pull request by bradfitz**
Updates tailscale/corp#8222

### CODE REVIEW

**Comment by bradfitz** [comment]
(not done testing this)

**Comment by bradfitz** [comment]
Okay, I think this is good now.

File: ipn/ipnserver/server.go [hunk 1]
```diff
@@ -63,22 +64,47 @@ func (s *Server) mustBackend() *ipnlocal.LocalBackend {
...
 }
 
+// waiterSet is a set of callers waiting on something. Each item (map value) in
+// the set is a func that wakes up that waiter's context. The waiter is responsible
+// for removing itself from the set when woken up. The (*waiterSet).add method
+// returns a cleanup method which does that removal. The caller than defers that
+// cleanup.
+//
+// TODO(bradfitz): this is a generally useful pattern. Move elsewhere?
+type waiterSet set.HandleSet[context.CancelFunc]
```
Comment (maisem) [inline ipn/ipnserver/server.go:74]: if the goal is to wake up everyone on an event, you could also do something like this and not have to worry about all the accounting and cleanup.

```
type waiterSet struct {
  ch syncs.AtomicValue[chan struct{}]
}
func (s *waiterSet) get() (ready <-chan struct{}) {
  return s.ch.Load()
}
func (s *waiterSet) wakeAll() {
  close(s.ch.Swap(make(chan struct{})))
}
```

Comment (maisem) [reply ipn/ipnserver/server.go:74]: you will need to initialize it once

File: ipn/localapi/localapi.go [hunk 2]
```diff
@@ -590,6 +591,32 @@ func (h *Handler) serveStatus(w http.ResponseWriter, r *http.Request) {
 	e.Encode(st)
 }
 
+// InUseOtherUserIPNStream reports whether r is a request for the watch-ipn-bus
+// handler. If so, it writes an ipn.Notify InUseOtherUser message to the user
+// returns true. Otherwise it returns false, in which case it doesn't write to
```
Comment (maisem) [inline ipn/localapi/localapi.go:596]: And returns true?

**Review by maisem — APPROVED** [review]

File: ipn/ipnserver/server.go (hunk 1 above)
Comment (creachadair) [reply ipn/ipnserver/server.go:74]: ^ I've likewise had good luck using this simpler pattern for the broadcast case (though I did it with explicit locks, https://pkg.go.dev/github.com/creachadair/msync#Trigger).

Comment (bradfitz) [reply ipn/ipnserver/server.go:74]: Per chat, we're going to do this later in its own PR adding the type/docs/tests and convert multiple callers.

File: ipn/localapi/localapi.go (hunk 2 above)
Comment (bradfitz) [reply ipn/localapi/localapi.go]: Done.

**Pull request merged**